index.html

```diff
@@ -2494,34 +2551,90 @@
-                <var>messageId</var> does not match the <a>URL pattern</a>
+                If <var>messageId</var> does not match the <a>URL pattern</a>
                 stored in <var>options</var>.url, then skip to the next
                 <a>NFC watch</a>.
               </li>
               <li>
-                If <var>options</var>.kind is defined, and it is not equal to
-                <var>record</var>.kind, then skip to the next <a>NFC watch</a>.
+                If <var>options</var>.kind is not equal to any
+                <var>record</var>.kind where <var>record</var> is an element of
+                <var>message</var>, then skip to the next <a>NFC watch</a>.
               </li>
               <li>
-                If <var>options</var>.type is defined, and it is not equal to
-                <var>record</var>.type, then skip to the next <a>NFC watch</a>.
+                If <var>options</var>.type is not equal to any
+                <var>record</var>.type where <var>record</var> is an element of
+                <var>message</var>, then skip to the next <a>NFC watch</a>.
               </li>
               <li>
                 <a>Queue a task</a> using the <dfn>Web NFC task source</dfn>
                 to invoke the <code>MessageCallback</code>
-                <var>callback</var> with <var>record</var> as its first
-                argument, and <var>messageId</var> as its second argument.
+                <var>callback</var> with <var>message</var> as its first
+                argument, and <var>messageId</var> as its second argument,
+                and then set the <var>receivedMessage</var> variable
+                <a href="#adapter-inbox">associated</a> with the current
+                <a>NFC adapter</a> be <code>null</code>.
               </li>
-            </ul>
+            </ol>
           </li>
-        </ul>
+        </ol>
+      </li>
+    </ol>
+    </section>
+  </section> <!-- receiving content -->
+
+  <section><h3>Handling visibility and focus</h3>
+    <p>
+      With the term <dfn id="nfc-suspended">suspended</dfn> this specification
+      refers to NFC operations being suspended, i.e. no <a>NFC content</a> is
+      pushed, and no received <a>NFC content</a> is presented via watches while
+      suspended.
+      However, timers instantiated on platform level for the
+      <code><a>NFCAdapter.pushMessage</a>()</code> method continue running,
+      and if they expire, the event should be recorded and handled
+      next time the execution is resumed, i.e. when the <code>focus</code>
+      event is fired on the <code>Window</code> object.
+    </p>
+    <p>
+      When the <code>Window</code> object associated with the
+      <code><a>Document</a></code> using the Web NFC API
+      <a href="http://www.w3.org/TR/page-visibility/#now-visible-algorithm">
+      becomes visible</a> and
+      <a href="https://html.spec.whatwg.org/#handler-onfocus">in focus</a>,
```

Comment:
It looks like https://html.spec.whatwg.org/#focus may be the best link. There's also https://html.spec.whatwg.org/#focused, but that only applies to elements, while you want this to apply to documents, windows, or browsing contexts.